niworkflows/utils/spaces.py

```diff
@@ -0,0 +1,497 @@
+    spec = attr.ib(factory=dict)
+    """The dictionary of specs."""
+    standard = attr.ib(default=False, repr=False, type=bool)
+    """Whether this space is standard or not."""
+    dim = attr.ib(default=3, repr=False, type=int)
+    """Dimensionality of the sampling manifold."""
+
+    def __attrs_post_init__(self):
+        """Extract cohort out of spec."""
+        if self.spec is None:
+            object.__setattr__(self, "spec", {})
+
+        if self.name.startswith('fsaverage'):
+            name = self.name
+            object.__setattr__(self, "name", "fsaverage")
+
+            if 'den' not in self.spec or name != "fsaverage":
+                spec = self.spec.copy()
+                spec['den'] = FSAVERAGE_DENSITY[name]
+                object.__setattr__(self, "spec", spec)
+
+        if self.name.startswith('fs'):
+            object.__setattr__(self, "dim", 2)
+
+        if self.name in self._standard_spaces:
+            object.__setattr__(self, "standard", True)
+
+        _cohorts = ["%s" % t
+                    for t in _tfapi.TF_LAYOUT.get_cohorts(template=self.name)]
+        if "cohort" in self.spec:
+            if not _cohorts:
+                raise ValueError(
+                    'standard space "%s" does not accept a cohort '
+                    'specification.' % self.name)
+
+            if str(self.spec["cohort"]) not in _cohorts:
+                raise ValueError(
+                    'standard space "%s" does not contain any cohort '
+                    'named "%s".' % (self.name, self.spec["cohort"]))
+        elif _cohorts:
+            _cohorts = ', '.join(['"cohort-%s"' % c for c in _cohorts])
+            raise ValueError(
+                'standard space "%s" is not fully defined.\n'
+                'Set a valid cohort selector from: %s.' % (self.name, _cohorts))
+
+    @property
+    def fullname(self):
+        """Generate a full-name combining cohort."""
+        if "cohort" not in self.spec:
+            return self.name
+        return "%s:cohort-%s" % (self.name, self.spec["cohort"])
+
+    @property
+    def legacyname(self):
+        """Generate a legacy name for fsaverageX spaces."""
+        if self.name == "fsaverage":
+            return FSAVERAGE_LEGACY[self.spec["den"]]
+
+    @name.validator
+    def _check_name(self, attribute, value):
+        if value.startswith('fsaverage'):
+            return
+        valid = list(self._standard_spaces) + NONSTANDARD_REFERENCES
+        if value not in valid:
+            raise ValueError(
+                'space identifier "%s" is invalid.\nValid '
+                'identifiers are: %s' % (value, ', '.join(valid)))
+
+    @spec.validator
+    def _check_spec(self, attribute, value):
+        if value is not None and not isinstance(value, dict):
+            raise TypeError(
+                "invalid space specification: %s." % str(value))
+
+    @classmethod
+    def from_string(cls, value):
+        """Create a new Space from string."""
```

Comment:
This seems to return a list of `Space`s.

Reply:
That is because it needs to generate lists of spaces to generate the 4 spaces of ``MNIPediatricAsym:cohort-1:cohort-2:res-1:res-2``.

Your comment is more about that it should be mentioned in the docstring? - I should be able to fix that :)

Reply:
Does this https://1088-74174878-gh.circle-artifacts.com/0/root/project/docs/_build/no_version_html/html/api/niworkflows.utils.spaces.html#niworkflows.utils.spaces.Space.from_string address your point?

Reply:
Yup, looks good.